File: src/me/wikyfg/ane/events/JoinEvent.java
package me.wikyfg.ane.events;

import me.wikyfg.ane.ANEMain;
import me.wikyfg.ane.files.Files;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerJoinEvent;


public class JoinEvent implements Listener {
    private ANEMain main;

    public JoinEvent(ANEMain main){
        this.main = main;
    }

    @EventHandler
    public void onJoin(PlayerJoinEvent e){
        Player p = e.getPlayer();
        if(Files.userdata.get(p.getName()) == null){
            Files.userdata.set(p.getName() + ".homes.home", "none");
            Files.userdata.set(p.getName() + ".back", p.getLocation().getX() + "," + p.getLocation().getY() + "," + p.getLocation().getZ() + "," + p.getLocation().getWorld().getName());
            main.files.saveFiles();
        }

    }

}

File: src/me/wikyfg/ane/events/JoinEvent.kt
package me.wikyfg.ane.events

import me.wikyfg.ane.ANEMain
import me.wikyfg.ane.api.ExperienceAPI
import me.wikyfg.ane.files.Files
import me.wikyfg.ane.utils.isInJail
import org.bukkit.Bukkit
import org.bukkit.ChatColor
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerJoinEvent

class JoinEvent(private val main: ANEMain) : Listener {
    @EventHandler
    fun onJoin(e: PlayerJoinEvent) {
        val p = e.player
        val expAPI = ExperienceAPI(p)
        p.apply {
            if (Files.userdata[name] == null) {
                Files.userdata["$name.homes.home"] = "none"
                Files.userdata["$name.back"] = "${location.x},${location.y},${location.z},${location.world?.name}"
                main.files?.saveFiles()
                Bukkit.broadcastMessage("${ChatColor.YELLOW}¡Bienvenido $displayName al servidor! ¡Recuerda usar /ane!")

            } else {
                p.sendMessage("${ChatColor.GREEN}¡Recuerda usar ${ChatColor.GOLD}/ane${ChatColor.GREEN}!")
            }
        }

        if (p.isInJail()) {
            Bukkit.getOnlinePlayers().forEach { player ->
                if (player.isOp) {
                    player.sendMessage("${ChatColor.GRAY}Ha entrado ${p.name}, que está encarcelado.")
                }
            }
        }
    }
}